pkg/restore/restore.go

```diff
@@ -578,6 +590,97 @@ func addToResult(r *api.RestoreResult, ns string, e error) {
 	}
 }
 
+func (ctx *context) waitForPVDeletion(name string, resourceClient client.Dynamic) (bool, error) {
+	pvLogger := ctx.log.WithField("pvName", name)
+
+	var shouldRestore bool
+	err := wait.PollImmediate(time.Second, ctx.resourceTerminatingTimeout, func() (bool, error) {
+		clusterPV, err := resourceClient.Get(name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			pvLogger.Debug("PV not found, safe to restore")
+			// PV not found, can safely exit loop and proceed with restore.
+			shouldRestore = true
+			return true, nil
+		}
+
+		if err != nil {
+			return true, errors.Wrapf(err, "could not retrieve in-cluster copy of PV %s", name)
+
+		}
+
+		var namespace, pvcName string
+		if clusterPV != nil {
```

Comment:
`clusterPV` should always be non-nil at this point, right? since we already returned if we got a not-found error. so can probably remove this `if`.  if you want to be safe, you could check that it's not nil and just return an error if it is, since it should never be the case.

Reply:
Removed this in the latest commit.